fe/fe-core/src/main/java/com/starrocks/sql/optimizer/function/MetaFunctions.java

```diff
@@ -461,4 +463,20 @@ public static ConstantOperator lookupString(ConstantOperator tableName,
         }
     }
 
+    /**
+     * Obtain the value of the latest partition that contains data.
+     */
+    @ConstantFunction(name = "max_partition", argTypes = {VARCHAR}, returnType = VARCHAR, isMetaFunction = true)
```

Comment:
1. i feel like using `last_partition` can be better, compared with `max_partition`, or `latest_partition`
2. Since this function returns the value of that partition, how could you proceed the requirement of returning partition name&id ? returning the partition id can make it composable with other functions

Reply:
i guess finally you would need to compose this function with other features, like `select value from table partition ($last$)`, so i would suggest decompose the function `max_function` to make to more composable rather than returning the lower point of partition directly

Reply:
Because the current user usage is: `select * from xxx where dt=(select max(dt) from xxx)`, which will result in a full table scan.
Because my idea is to implement the `max_partition` function and return the partition value, so that `dt=max_partition(table_name)`, the full table will not be scanned.
But I realized that if the range of a partition is multiple days, the query result will not meet expectations.
It is a good way to specify the partition of the query through `partition(partition_name)`. But currently `partition(last_partition(table_name))` cannot be executed. It must be a partition name, not a function.
So do you have any better suggestions?